### ISSUE TITLE
Add support for type checking for the equality operator

### ISSUE THREAD

**Issue by louis-van-der-stam** [python/mypy#12410]
**Feature**

```
def silly_sample1()-> bool:
    return 123 == 'something'  # Complain here about comparing different types of objects

def silly_sample2()-> bool:
    return 'Something'.lower == 'something'  # Complain here about comparing different types of objects

def silly_sample3()-> bool:
    return 'Something'.lower() == 'something'  # Works


```
**Pitch**

It would be really nice if these basic comparison operators would be capable of detecting these kinds of errors.

**Comment by louis-van-der-stam** [comment]
That’s really great! Is this under control of some configuration flag? For me this obviously does not happen….

**Comment by KotlinIsland** [comment]
Sure, `strict = true`, but more specifically it's `strict-equality`

**Comment by louis-van-der-stam** [comment]
Thanks didn't expect this to be disabled by default. Will check the other options also!

Thanks for pointing it out!

**Comment by KotlinIsland** [comment]
I would also recommend `enable_error_codes = "truthy-bool, redundant-expr"`

**Comment by KotlinIsland** [comment]
@louis-van-der-stam There is a fork of mypy called basedmypy, where these options are all enabled by default.

**Comment by JukkaL** [comment]
Some of these strictness options aren't enabled by default because they can (or at least used to) generate false positives in some cases. Strict equality is mature enough now that we could consider enabling it by default.
